store/mongo/datastore_mongo_cleanup.go

```diff
@@ -104,29 +117,33 @@ func (db *DataStoreMongo) GetBrokenAuthSets(dbName string) ([]string, error) {
 // Get Ids of the tokens owned by devices that are in decommissioning state and tokens not
 // owned by any device.
 func (db *DataStoreMongo) GetBrokenTokens(dbName string) ([]string, error) {
+	c := db.client.Database(dbName).Collection(DbTokensColl)
 
-	s := db.session.Copy()
-	defer s.Close()
+	ctx := context.Background()
 
 	deviceIds := []string{}
 	brokenTokens := []string{}
-	c := s.DB(dbName).C(DbTokensColl)
 
 	// get all tokens; group by device id
 
-	job := &mgo.MapReduce{
-		Map:    "function() { emit(this.dev_id, 1) }",
-		Reduce: "function(key, values) { return Array.sum(values) }",
+	group := bson.D{
+		{Key: "$group", Value: bson.D{
+			{Key: "_id", Value: "$" + "dev_id"}},
```

Comment:
Same as above - why the string concatenation?

Reply:
fixed